cms/djangoapps/contentstore/models.py

```diff
@@ -201,7 +204,10 @@ def summarize_by_downstream_context(cls, downstream_context_key: CourseKey) -> Q
             upstream_context_title=F("upstream_block__publishable_entity__learning_package__title"),
         ).annotate(
             ready_to_sync_count=Count("id", Q(ready_to_sync=True)),
-            total_count=Count('id')
+            total_count=Count("id"),
+            last_published_at=Max(
+                "upstream_block__publishable_entity__published__publish_log_record__publish_log__published_at"
```

Comment:
@bradenmacdonald @ormsbee Is there a better way to get the published date for an upstream component here?

Reply:
Not that I'm aware of. Though I think you could do a much more efficient query to just check when the particular upstream learning context (library) was last published, without caring about whether or not that publish was used in upstream-downstream links.